presto-main/src/main/java/com/facebook/presto/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -223,7 +223,7 @@ public void onFailure(Throwable t)
                 splitAssignment = splitPlacementResult.getAssignments();
 
                 // remove splits with successful placements
-                pendingSplits.removeAll(splitAssignment.values());
+                splitAssignment.values().forEach(pendingSplits::remove); // AbstractSet.removeAll performs terribly here.
```

Comment:
`pendingSplits` is a always a `j.u.HashSet`. Does it mean `removeAll()` is generally not safe to use?

Reply:
Oh, nvrm